controllers/awsmachine_controller.go

```diff
@@ -1178,3 +1186,22 @@ func (r *AWSMachineReconciler) ensureInstanceMetadataOptions(ec2svc services.EC2
 
 	return ec2svc.ModifyInstanceMetadataOptions(instance.ID, machine.Spec.InstanceMetadataOptions)
 }
+
+// +kubebuilder:rbac:groups=controlplane.cluster.x-k8s.io,resources=*,verbs=get;list;watch
```

Comment:
TBH, this wildcard makes obsolete the following ones:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/6193d172e0887d08319a63e07720ac9f7f08a841/controllers/rosacluster_controller.go#L55
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/6193d172e0887d08319a63e07720ac9f7f08a841/bootstrap/eks/controllers/eksconfig_controller.go#L62
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/6193d172e0887d08319a63e07720ac9f7f08a841/controllers/awsmanagedcluster_controller.go#L54
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/6193d172e0887d08319a63e07720ac9f7f08a841/controllers/rosacluster_controller.go#L55
...and many other lines.

Although we could optimize the role rules, I think these  are more intuitive for developers, since it's pretty straightforward trying to understand the objects controlled by the various controllers.

Reply:
Let's move these to a single place above